pantheon-bundle/src/main/java/com/redhat/pantheon/extension/HydraIntegration.java

```diff
@@ -104,11 +113,26 @@ public void processEvent(Event event) throws Exception {
         MessageProducer producer = session.createProducer(session.createTopic(HYDRA_TOPIC));
         String moduleUUID = module.getValueMap().get(UUID_FIELD, String.class);
         String eventValue = ModuleVersionPublishedEvent.class.equals(event.getClass()) ? EVENT_PUBLISH_VALUE : EVENT_UNPUBLISH_VALUE;
-        String msg = "{\""
-                + ID_KEY + "\":" + "\"" + this.getPantheonHost() + PANTHEON_MODULE_API_PATH + moduleUUID +"\","
-                + "\"" + EVENT_KEY + "\":" + "\"" + eventValue + "\"}";
+        String idValue = ModuleVersionPublishedEvent.class.equals(event.getClass()) ? this.getPantheonHost() + PANTHEON_MODULE_API_PATH
+                + moduleUUID : "";
+        String uriValue = "";
+        String msg = "";
+        if (System.getenv(PORTAL_URL) != null) {
```

Comment:
what  happens if the portal URL is not defined? seems like it would still seek to publish an empty id.
Should we abort the message in that case?

Reply:
good catch. i've refactored the code.